### PR TITLE
CI: shellcheck tools/* and add a shellcheck-tip tox environment

### PR DESCRIPTION
**Pull request by paride**

### CODE REVIEW

**Comment by paride** [comment]
@OddBloke I resubmitted again (third time) this PR and now CI ran without changing anything. I think we hit a GitHub hiccup.

File: tox.ini [hunk 1]
```diff
@@ -36,8 +46,6 @@ commands = python3 -m flake8 {[common]pyfiles}
 deps = pylint
 commands = python3 -m pylint {[common]pyfiles}
 
-[testenv:shellcheck]
-# shellcheck from Xenial is too old and doesn't support the --severity
-# argument. Let's pull the latest version using shellcheck-py.
+[testenv:shellcheck-tip]
```
Comment (OddBloke) [inline tox.ini:49]: For consistency with the tip environments in cloud-init and curtin:

```suggestion
[testenv:tip-shellcheck]
```

Comment (paride) [reply tox.ini:49]: `pylint-tip` and `flake8-tip` are to be changed too. The intention *was* to have the same envs that we have in curtin and cloud-init, but I got them wrong.

**Comment by paride** [comment]
TIL: https://github.com/rpkilby/tox-factor

tox plugin that allows running the tox envs matching a pattern, e.g. `tox -f tip` should works. I say should as I didn't really try :)

**Pull request merged**